polkadot/xcm/src/v5/traits.rs

```diff
@@ -305,11 +305,11 @@ pub trait ExecuteXcm<Call> {
 	) -> Outcome {
 		let pre = match Self::prepare(message) {
 			Ok(x) => x,
-			Err(_) => return Outcome::Error { error: Error::WeightNotComputable },
+			Err(_) => return Outcome::Error { error: Error::WeightNotComputable, index: 0 },
 		};
 		let xcm_weight = pre.weight_of();
 		if xcm_weight.any_gt(weight_limit) {
-			return Outcome::Error { error: Error::WeightLimitReached(xcm_weight) }
+			return Outcome::Error { error: Error::WeightLimitReached(xcm_weight), index: 0 }
```

Comment:
need to fix the `index` here, it should say which instruction took the message over the weight limit